### DISCUSSION TITLE
Lazy relations loaded via injected request

### DISCUSSION THREAD

**Discussion by thewebartisan7** [spatie/laravel-data#212] [General]
Supposely we have 

```php
    public static function fromModel(Model $model): self
    {
        return new self(
            relations: Lazy::whenLoaded('relations', $model, fn () => ModelData::collection($model->relations)),
            relationsCount: Lazy::create(fn () => $model->relations_count)
        );
    }
```

How to avoid loading `relations` when in query we have `withCount('relations')` ?

Because in my case it load also the `relations` when `withCount('relations')` is loaded.

**Comment by Nir-An** [comment]
Hi @thewebartisan7 
what happens if you do something like : 

```php
relations: Lazy::whenLoaded('relations', $model, fn () => ModelData::collection($model->relations)),
relationsCount: Lazy::whenLoaded('relations', $model, fn () => $model->relations_count)

```
?

**Reply by thewebartisan7** [reply]
It's the same, because I check RelationLazy and behind the scene it's just check if relations is loaded, and in both case, relations_count and relations is marked as loaded:

```php
    public function shouldBeIncluded(): bool
    {
        return $this->model->relationLoaded($this->relation);
    }
```

**Comment by Nir-An** [comment]
Hi @rubenvanassche 
from  the first view, I think the problem is here :
https://github.com/spatie/laravel-data/blob/main/src/Transformers/DataTransformer.php#L73

```php
  if (! $this->shouldIncludeProperty($name, $data->{$name}, $trees)) {
                    return $payload;
  }
```
```php
$data->{$name}

```
but I'm not sure, I'm going to test it

**Comment by rubenvanassche** [comment]
Could you provide me a test which reproduces the problem? Tried this by changing this test (which may be a good starting point):

https://github.com/spatie/laravel-data/blob/782aced6741d6b5bd45e372a439326c4ba971da9/tests/DataTest.php#L290-L303

But couldn't reproduce the issue

**Reply by thewebartisan7** [reply]
I will try to reproduce in a test, but not sure when get time.
This was just a test project that I was working on, and currently is only locally.
Give me a week or two.
